jest/assetFileTransformer.js

```diff
@@ -8,12 +8,18 @@
  */
 'use strict';
 
+/* eslint-env node */
+
 const path = require('path');
+const createCacheKeyFunction = require('fbjs-scripts/jest/createCacheKeyFunction');
 
 module.exports = {
   // Mocks asset requires to return the filename. Makes it possible to test that
-  // the correct images are loaded for components. So require('img1.png')
-  // becomes 'img1.png' in the Jest snapshot.
+  // the correct images are loaded for components. Essentially
+  // require('img1.png') becomes `Object { "uri": 'path/to/img1.png' }` in the
+  // Jest snapshot.
   process: (_, filename) =>
-    `module.exports = ${JSON.stringify(path.basename(filename))};`
+    `module.exports = {uri: ${JSON.stringify(path.relative('.', filename))}};`,
+
+  getCacheKey: createCacheKeyFunction([__filename]),
```

Comment:
This function was needed after rebasing on latest master. I took some inspiration from https://github.com/facebook/react-native/blob/master/jest/preprocessor.js.

Without it running Jest will give the following error:

```
 FAIL  Libraries/Image/__tests__/assetBaseNameInSnapshot.js
  ● renders assets based on basename

    /Users/martin/git/react-native/Libraries/Image/__tests__/img/img1.png:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){module.exports = [object Object];
                                                                                                                      ^^^^^^

    SyntaxError: Unexpected identifier

      at ScriptTransformer._transformAndBuildScript (node_modules/jest-runtime/build/script_transformer.js:305:17)
      at Object.<anonymous> (Libraries/Image/__tests__/assetBaseNameInSnapshot.js:21:64)
      at tryCallTwo (node_modules/promise/lib/core.js:45:5)
      at doResolve (node_modules/promise/lib/core.js:200:13)

  ✕ renders assets based on basename (144ms)
```